### PR TITLE
Add network profile

### PR DESCRIPTION
**Pull request by nicholasmhughes**
### What does this PR do?
Add network profiles. Needed for container instance use of vnets

### What issues does this PR fix or reference?
Fixes: N/A

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to idem-azurerm require tests.**
- [x] Docs
- [x] Tests written/updated

### Commits signed with GPG?
Yes

Please review [idem-azurerm's Contributing Guide](https://github.com/eitrtechnologies/idem-azurerm/blob/master/.github/CONTRIBUTING.md) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: idem_azurerm/exec/azurerm/network/network_profile.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+    :param container_network_interface_configurations: List of chid container network interface configurations.
+
+    :param tags: The tags of the resource.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.network.network_profile.update myprofile mygroup
```
Comment (Ajnbro) [inline idem_azurerm/exec/azurerm/network/network_profile.py:90]: This should be create_or_update

File: idem_azurerm/exec/azurerm/network/network_profile.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+    Creates or updates a network profile.
+
+    :param name: The name of the network profile.
+
+    :param resource_group: The name of the resource group to which the network profile belongs.
+
+    :param container_network_interfaces: List of child container network interfaces.
+
+    :param container_network_interface_configurations: List of chid container network interface configurations.
```
Comment (Ajnbro) [inline idem_azurerm/exec/azurerm/network/network_profile.py:82]: child

File: idem_azurerm/states/azurerm/network/network_profile.py [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+    Ensure a network profile exists.
+
+    :param name: The name of the network profile.
+
+    :param resource_group: The name of the resource group to which the network profile belongs.
+
+    :param container_network_interfaces: List of child container network interfaces.
+
+    :param container_network_interface_configurations: List of chid container network interface configurations.
```
Comment (Ajnbro) [inline idem_azurerm/states/azurerm/network/network_profile.py:105]: child

File: idem_azurerm/states/azurerm/network/network_profile.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    """
+    .. versionadded:: 3.0.0
+
+    Ensure a network profile exists.
+
+    :param name: The name of the network profile.
+
+    :param resource_group: The name of the resource group to which the network profile belongs.
+
+    :param container_network_interfaces: List of child container network interfaces.
```
Comment (Ajnbro) [inline idem_azurerm/states/azurerm/network/network_profile.py:103]: Should we add more information regarding what specifically needs to be passed (i.e., a dictionary with the key value pairs 1, 2, and 3, etc.)? This is more of a suggestion to help avoid a user getting confused or not realized what parameters are fully required.

File: tests/integration/states/azurerm/network/test_network_profile.py [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    }
+    ret = await hub.states.azurerm.network.network_profile.present(
+        ctx, prf, resource_group
+    )
+    assert ret == expected
+
+
+@pytest.mark.run(after="test_present", before="test_absent")
+@pytest.mark.asyncio
+async def test_changes(hub, ctx, resource_group, acr, tags):
```
Comment (Ajnbro) [inline tests/integration/states/azurerm/network/test_network_profile.py:25]: acr isn't used here.

File: idem_azurerm/states/azurerm/network/network_profile.py (hunk 4 above)
Comment (nicholasmhughes) [reply idem_azurerm/states/azurerm/network/network_profile.py]: they are really complex structures with nested objects... and I don't see people using this state very much. I'll link the SDK docs to give direction and I'll cover it in my "how to containers" topic for the docs.

**Pull request merged**